### ISSUE TITLE
false negative unused_mut

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust#57248]
````rust
struct SomeStruct {
	path: std::path::PathBuf,
}

impl SomeStruct {
	fn path_exists(&mut self) -> bool { // false negative unused_mut
		// path.exists() does not need mutable reference:
		// pub fn exists(&self) -> bool { ... } 
		self.path.exists()
	}
}

fn main() {
    let mut s = SomeStruct { path: std::path::PathBuf::from("/tmp/a"), };
    println!("{}", s.path_exists());
}
````

**Comment by zackmdavis** [comment]
I'm pretty sure the unused-mut lint is specifically about the `mut` mutability modifier keyword (as in `let mut a =` ...). Changing `&mut` to `&` where possible sounds like a candidate Clippy lint?

**Comment by scottmcm** [comment]
I think this is by design, since that's not a `let mut`, and there are potential reasons for an API to take `&uniq` even if it only calls shared stuff (or nothing).

**Comment by zackmdavis** [comment]
jinx

**Comment by sanxiyn** [comment]
This is rust-lang/rust-clippy#1230. I agree it is out of scope for built-in unused_mut lint. Closing.
